tests/curators/test_curators_examples.py

```diff
@@ -214,9 +215,21 @@ def test_dataframe_curator(small_dataset1_schema: ln.Schema, ccaplog):
         assert str(error).startswith("column 'sample_note' not in dataframe")
     curator.standardize()
     curator.validate()
+
     artifact.delete(permanent=True)
 
 
+def test_dataframe_curator_index():
+    """Test validating a DataFrame index."""
+    df = datasets.small_dataset1(otype="DataFrame", with_index_type_mismatch=True)
+    schema = ln.Schema(index=ln.Feature(name="test", dtype="str").save()).save()
+    curator = ln.curators.DataFrameCurator(df, schema)
+    try:
```

Comment:
Sorry, was dragged into another meeting before.

We need `pytest.raises()` here so that we're sure the error is actually raised.